server/src/main/java/org/opensearch/common/unit/Fuzziness.java

```diff
@@ -192,6 +195,9 @@ public static Fuzziness parse(XContentParser parser) throws IOException {
                     // Validate if the fuzziness value is formatted correctly as a numeric value.
                     try {
                         final float minimumSimilarity = Float.parseFloat(fuzziness);
+                        if (minimumSimilarity < 0.0f && !Float.isInfinite(minimumSimilarity) && !Float.isNaN(minimumSimilarity)) {
```

Comment:
Hm ... it looks like the condition is not correct? Shouldn't it be:

```suggestion
                        if (minimumSimilarity < 0.0f || Float.isInfinite(minimumSimilarity) || Float.isNaN(minimumSimilarity)) {
```

Reply:
You're absolutely correct... Sorry, super silly mistake on my part. Completely flew by me 😅